File: backend/src/main/java/com/nexadine/service/PaymentService.java
package com.nexadine.service;

import org.json.JSONObject;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Service;

import com.nexadine.dto.PaymentResponse;
import com.nexadine.dto.PaymentVerificationRequest;
import com.razorpay.Order;
import com.razorpay.RazorpayClient;

@Service
public class PaymentService {

    @Value("${razorpay.key.id}")
    private String keyId;

    @Value("${razorpay.key.secret}")
    private String keySecret;

    // Create Razorpay Order
    public Order createOrder(Double amount) throws Exception {

        if (amount == null || amount <= 0) {
            throw new RuntimeException("Invalid payment amount.");
        }

        RazorpayClient razorpay = new RazorpayClient(keyId, keySecret);

        JSONObject orderRequest = new JSONObject();

        // Amount in paise
        orderRequest.put("amount", (int) Math.round(amount * 100));

        orderRequest.put("currency", "INR");

        orderRequest.put(
                "receipt",
                "ORDER_" + System.currentTimeMillis());

        orderRequest.put("payment_capture", 1);

        return razorpay.orders.create(orderRequest);
    }

    // Verify Payment
    public PaymentResponse verifyPayment(PaymentVerificationRequest request) {

        System.out.println("========== PAYMENT VERIFY ==========");
        System.out.println("Order ID    : " + request.getRazorpay_order_id());
        System.out.println("Payment ID  : " + request.getRazorpay_payment_id());
        System.out.println("Signature   : " + request.getRazorpay_signature());
        System.out.println("====================================");

        // Demo mode: skip signature verification
        return new PaymentResponse(
                true,
                "Payment Verified Successfully");
    }

}

File: frontend/src/services/paymentService.js
import axios from "axios";

const API_URL = "http://localhost:8080/api/payment";

// Create Razorpay Order
export const createOrder = async (amount) => {
  try {
    const response = await axios.post(
      `${API_URL}/create-order`,
      {
        amount,
      }
    );

    return response.data;
  } catch (error) {
    console.error("Create Order Error:", error);
    throw error;
  }
};

// Verify Razorpay Payment
export const verifyPayment = async (paymentData) => {
  try {
    const response = await axios.post(
      `${API_URL}/verify`,
      paymentData
    );

    return response.data;
  } catch (error) {
    console.error("Verify Payment Error:", error);
    throw error;
  }
};